Formula/l/logtalk.rb

```diff
@@ -25,4 +25,12 @@ class Logtalk < Formula
   def install
     cd("scripts") { system "./install.sh", "-p", prefix }
   end
+
+  def caveats
+    <<~EOS
+      To complete the Logtalk installation, define the environment variables:
+        LOGTALKHOME=#{HOMEBREW_PREFIX}/share/logtalk
+        LOGTALKUSER=$HOME/logtalk
+    EOS
+  end
```

Comment:
Is this not in the upstream documentation?

Reply:
`caveats` only exist to document Homebrew specific caveats. Not to document upstream behaviour. What might be better is adding a `env_script` to automatically set these, so people don't have to care.

Reply:
As I wrote in my previous reply, the requirement to define those two environment variables **is** documented upstream. With other Logtalk installers (including the Windows installer and the macOS installer generated using MacPorts), those variables are added automatically. Thus, this issue is specific to Homebrew packaging. That fits squarely the documentation of `caveats` as described in the Homebrew "Formula Cookbook". I assume that's why @chenrui333 recommended its use instead of `ohai`.

That said, I would prefer a solution to automatically set the environment variables. I see both `env_script_all_files` and `write_env_script`. Both are documented as:

> This method is part of a private API. This method may only be used in the [Homebrew/brew](https://github.com/Homebrew/brew) repository. Third parties should avoid using this method if possible, as it may be removed or changed without warning.

But neither is appropriate in my case. There isn't a single binary but a set of shell scripts that integrate Logtalk with supported Prolog backends. In the case, the MacPorts generated installer, a postflight script writes to `/etc/zprofile`, `/etc/profile`, and `/etc/csh.cshrc`. Planning to do the same for `brew` installs at some later time. Unless there's an alternative solution that you can recommend?

Reply:
@chenrui333 @SMillerDev Can we maker progress here? A new Logtalk version is expected shortly. Thanks.